tap_slack/streams.py

```diff
@@ -32,9 +32,26 @@ def get_url_params(self, context, next_page_token):
     def post_process(self, row, context):
         "Join the channel if not a member, but emit no data."
         row = super().post_process(row, context)
-        if not row["is_member"]:
-            self._join_channel(row["id"])
-        return row
+        # return all in selected_channels or default to all, exclude any in excluded_channels list
+        channel_name = row["id"]
+        if self._is_included(channel_name) and not self._is_excluded(channel_name):
+            if not row["is_member"] and self.config.get("auto_join_channels", False):
+                self._join_channel(channel_name)
+            return row
+
+    def _is_excluded(self, channel_name: str) -> bool:
+        excluded_channels = self.config.get("excluded_channels", [])
+        if channel_name in excluded_channels:
+            return True
+        else:
+            return False
+
+    def _is_included(self, channel_name: str) -> bool:
```

Comment:
Could this function evaluate both the include/exclude functionality? Also, naming might be clearer if it's `_is_channel_inluded`, since the stream class itself has some properties like `selected`, etc.

Reply:
Yeah I had that but it was getting a little hard to read so I split them. I gave it another go and I think it looks decently clear now. Let me know what you think!

Reply:
Added an optional suggestion to make it even clearer, but it's your call on whether to accept it. Thanks!